pkg/streamformatter/streamformatter.go

```diff
@@ -170,3 +188,22 @@ func (sf *StderrFormatter) Write(buf []byte) (int, error) {
 	}
 	return len(buf), err
 }
+
+// AuxFormatter is a streamFormatter that writes aux progress messages
+type AuxFormatter struct {
+	io.Writer
+	*StreamFormatter
```

Comment:
@dnephin OK, I made that change. I considered adding `NewAuxFormatter` for consistency with 
#32949, but I didn't in the end because `AuxFormtter` wants to remain a public type since unlike `NewStd*Formatter ` it isn't just an `io.Writer`.

I could add it anyway if you prefer, or even introduce an `Emitter` interface if you would like to hide the implementation details even more.